lib/temp-directory.js

```diff
@@ -9,12 +9,32 @@ var osenv = require('osenv');
 
 function create(context, next) {
   var _path = path.join(osenv.tmpdir(),uuid.v4());
-  var npmConfigTmp = path.join(_path, 'npm_config_tmp');
-
-  context.path = _path;
+  var npmConfigTmp = _path;
+  // Overwrite the npm temp directory if not already set via the env variable
+  if (!process.env.npm_config_tmp) {
+    // Use the same directory as --tmpDir for npm if set
```

Comment:
`--tmpDir`-> `the citgm temp directory` to make it clear that tmpDir is the Citgm temp directory